Source/WebCore/layout/formattingContexts/inline/InlineLineBuilder.cpp

```diff
@@ -1078,6 +1118,40 @@ bool LineBuilder::tryPlacingFloatBox(const InlineItem& floatItem, LineBoxConstra
     auto& floatBox = floatItem.layoutBox();
     ASSERT(formattingState());
     auto& boxGeometry = formattingState()->boxGeometry(floatBox);
+    auto containingBlockMarginTrim = floatBox.parent().style().marginTrim();
```

Comment:
This does not look correct for cases when the float is nested in an inline box. margin-trim is set on block containers  (at least in this context, ignoring flex, grid etc) and inside an IFC by definition you won't be finding a block container (or containing block). This needs to consult with the root of the IFC.

Reply:
(please add a test case that cover this specific case as well)